packages/main/src/Toolbar.ts

```diff
@@ -0,0 +1,562 @@
+	 * @public
+	 */
+	@property({ type: ToolbarStyling, defaultValue: "Default" })
+	styling!: `${ToolbarStyling}`;
+
+	/**
+	 * Defines if the toolbar overflow popup should close upon intereaction with the items inside.
+	 * It will close by default.
+	 * @type {boolean}
+	 * @public
+	 * @name sap.ui.webc.main.Toolbar.prototype.preventOverflowClosing
+	 */
+	@property({ type: Boolean })
+	preventOverflowClosing!: boolean;
+
+	/**
+	 * Indicated the direction in which the Toolbar items will be aligned.
+	 * Available options are:
+	 * <ul>
+	 * <li><code>End</code></li>
+	 * <li><code>Start</code></li>
+	 * </ul>
+	 * @type {sap.ui.webc.main.types.ToolbarAlign}
+	 * @public
+	 * @defaultvalue: "End"
+	 * @name sap.ui.webc.main.Toolbar.prototype.alignContent
+	 */
+	@property({ type: ToolbarAlign, defaultValue: "End" })
+	alignContent!: `${ToolbarAlign}`;
+
+	/**
+	 * Items, which will be displayed in the toolbar.
+	 * @type {Object}
+	 * @private
+	 */
+	@property({ type: Object, multiple: true })
+	itemsToBar!: Array<ToolbarItem>;
+	/**
+	 * Items, that will be displayed inside overflow Popover.
+	 * @type {Object}
+	 * @private
+	 */
+	@property({ type: Object, multiple: true })
+	itemsToOverflow!: Array<ToolbarItem>;
+
+	/**
+	 * Cached the sum of all of items width.
+	 * @type {Object}
+	 * @private
+	 */
+	 @property({ type: Integer })
+	 ITEMS_WIDTH: number;
+
+	/**
+	 * Indicates the items have been measured and the layout can be calculated.
+	 * @type {boolean}
+	 * @private
+	 */
+	@property({ type: Boolean })
+	itemsWidthMeasured!: boolean;
+	/**
+	 * Indicates the end of the resizing iteration.
+	 * @type {boolean}
+	 * @private
+	 */
+	@property({ type: Boolean })
+	resizing!: boolean;
+
+	/**
+	* Calculated width of the whole toolbar.
+	* @private
+	*/
+	@property({ type: Integer })
+	width!: number;
+
+	/**
+	* @private
+	* Calculated width of all the Toolbar items.
+	*/
+	@property({ type: Integer })
+	contentWidth!: number;
+
+	/**
+	 * Notifies the toolbar if it should show the items in a reverse way if Toolbar Popover needs to be placed on "Top" position.
+	 * @private
+	 * @type {Boolean}
+	 */
+	@property({ type: Boolean })
+	reverseOverflow!: boolean;
+
+	/**
+ 	* Slotted Toolbar items
+	* @type {sap.ui.webc.main.IToolbarItem[]}
+	* @name sap.ui.webc.main.Toolbar.prototype.items
+	* @slot items
+	* @public
+	*/
+	@slot({ "default": true, type: HTMLElement, invalidateOnChildChange: true })
+	items!: Array<ToolbarItem>
+
+	_onResize!: ResizeObserverCallback;
+
+	ITEMS_WIDTH_MAP: Map<string, number>;
+
+	constructor() {
+		super();
+
+		// indicates if the bar is being reized at the moment
+		this.resizing = false;
+
+		this.ITEMS_WIDTH_MAP = new Map();
+
+		this.ITEMS_WIDTH = 0;
+
+		// resize handler
+		this._onResize = this.onResize.bind(this);
+	}
+
+	get OVERFLOW_BTN_SIZE(): number {
+		const toolbarOverflowButton = this.shadowRoot!.querySelector(".ui5-tb-overflow-btn")!;
```

Comment:
you should not use `!` for querySelector because the overflow button is conditionally rendered and it might miss in some scenarios.

and you can replace that query selector with overflowButtonDOM getter